src/common/datas/view.ts

```diff
@@ -0,0 +1,61 @@
+import { DataHolder } from './holder';
+
+/**
+ * Creates a typed view into a data holder. The view contains values that were
+ * in the data holder when this was called. New values assigned to the view are
+ * immediately reflected to the data holder. Mutations to values in view are
+ * not updated to the data holder; immutable values should be preferred.
```

Comment:
> Mutations to values in view are
not updated to the data holder; immutable values should be preferred.

While this is in theory good idea, my experience is that in a lot of real world scenarios the code will be a lot cleaner if we just allow mutating the objects directly. Compare
```ts
view.a = { ...view.a, b: { ...view.a.b, c: 1 } }
```
to
```ts
view.a.b.c = 1
```

Though obviously everything is fine as long as the highest level api (Blocks, CustomItems...) allows this kind of pattern

Reply:
After major implementation changes, this restriction has been lifted.